### DISCUSSION TITLE
SDK Transaction object creation from proto Transaction Body

### DISCUSSION THREAD

**Discussion by aadityapaliwal** [hashgraph/hedera-sdk-java#1777] [Q&A]
Hello everyone

In our Android Wallet, we are using Hedera dApp to use SignTransaction method. I am getting the request but I am not able to create SDK Transaction class object (Because it's abstract class) or child class object using the proto TransactionBody. Can anyone help me? I am using Java SDK (com.hedera.hashgraph:sdk:2.27.0)

val protoTxnBody = com.hedera.hashgraph.sdk.proto.TransactionBody.parseFrom(signData.txnBytes)
com.hedera.hashgraph.sdk.Transaction(protoTxnBody)  //Cannot create an instance of an abstract class
com.hedera.hashgraph.sdk.TransferTransaction(protoTxnBody) //Cannot access '<init>': it is package-private in 'TransferTransaction'

Is there anything that I am missing?

**Comment by thenswan** [comment]
Hi @aadityapaliwal, please check `fromBytes` and `fromProtobuf` static methods in transactions; based on your intention, it might be what you need. Let me know if it helps 🙂

**Reply by aadityapaliwal** [reply]
@thenswan `fromBytes` needs transaction bytes but in this case I am getting  `transactionBody` : `String` - Base64 encoded string representation of TransactionBody. I didn't find `fromProtobuf` in the **Transaction** class.

To give you more context, I am trying to use hedera_signtransaction method of Hedera dApp https://specs.walletconnect.com/2.0/blockchain-rpc/hedera-rpc#hedera_signtransaction

**Reply by aadityapaliwal** [reply]
@thenswan Any thoughts?

**Reply by thenswan** [reply]
@aadityapaliwal, thanks for the additional info. I reviewed this issue and would like to ask for MCVE on GitHub, where the issue can be verified.
Without MCVE now, I can recommend the following solution that might help to get the correct Base64 encoded string representation of TransactionBody:
- use `toBytes()` method on the transaction
- convert bytes to  Base64 encoded string

Please let me know if that helps :)

**Reply by aadityapaliwal** [reply]
@thenswan I will check if I can provide a MCVE but I think it's not any issue. My real problem is - 

- I am getting Base64 String of TransactionBody from dApp itself in the signTransaction method.
- Now, I have to sign this transaction and return it to the dApp.
- The dApp is expecting a signature Map (Base64 encoded string of SignatureMap)

So the issue I am facing is I am not able to convert Base64 TransactionBody into Transaction object. So that I can display the complete information to user on wallet and sign it. :(

**Reply by thenswan** [reply]
@aadityapaliwal, you can try to provide minimal example of flow with your comments - that would help.

Based on current info I can advice trying the following:
- decode Base64 String
- convert it to bytes
- use `Transaction.fromBytes()` method
